backend/inspirehep/records/marshmallow/common/mixins.py

```diff
@@ -17,3 +20,29 @@ def set_can_edit(data, orginal_data):
         if is_superuser_or_cataloger_logged_in():
             data["can_edit"] = True
         return data
+
+
+class CanEditByCollectionPermissionMixin:
+    def __init__(self, *args, **kwargs):
+        super().__init__()
+        self.post_dumps.append(self.set_can_edit)
+
+    @staticmethod
+    def set_can_edit(data, orginal_data):
```

Comment:
can you have somehow only a single permission checking logic, and reuse it here, instead of doing the same thing twice?